libs/api/ebpf_api.cpp

```diff
@@ -1814,14 +1811,68 @@ _ebpf_pe_get_map_count(
-        // which starts with a 16-byte-aligned NULL pointer.
+        // which starts with an 8-byte-aligned NULL pointer where the previous
+        // byte (if any) is also 00, and the following 8 bytes are non-NULL.
         uint32_t map_offset = 0;
         uint64_t zero = 0;
-        while (map_offset < section_header.Misc.VirtualSize &&
-               memcmp(buffer->buf + map_offset, &zero, sizeof(zero)) != 0) {
-            map_offset += 16;
+        while (map_offset + 16 < section_header.Misc.VirtualSize &&
+               (memcmp(buffer->buf + map_offset, &zero, sizeof(zero)) != 0 ||
+                (map_offset > 0 && buffer->buf[map_offset - 1] != 0) ||
+                memcmp(buffer->buf + map_offset + 8, &zero, sizeof(zero)) == 0)) {
+            map_offset += 8;
+        }
+        if (pe_context->object != nullptr) {
+            for (int map_index = 0; map_offset + sizeof(map_entry_t) <= section_header.Misc.VirtualSize;
+                 map_offset += sizeof(map_entry_t), map_index++) {
+                map_entry_t* entry = (map_entry_t*)(buffer->buf + map_offset);
+
+                ebpf_map_t* map = (ebpf_map_t*)calloc(1, sizeof(ebpf_map_t));
+                if (map == nullptr) {
+                    return 1;
+                }
+
+                map->map_handle = ebpf_handle_invalid;
+                map->original_fd = (fd_t)map_index;
+                map->map_definition.type = entry->definition.type;
+                map->map_definition.key_size = entry->definition.key_size;
+                map->map_definition.value_size = entry->definition.value_size;
+                map->map_definition.max_entries = entry->definition.max_entries;
+                map->map_definition.pinning = entry->definition.pinning;
+                map->map_definition.inner_map_id = entry->definition.inner_id;
+                map->inner_map_original_fd = entry->definition.inner_map_idx;
+                map->pinned = false;
+                map->reused = false;
+                map->pin_path = nullptr;
+
+                const char* map_name =
+                    _ebpf_get_section_string(pe_context, (uintptr_t)entry->name, section_header, buffer);
+                map->name = _strdup(map_name);
+                if (map->name == nullptr) {
+                    clean_up_ebpf_map(map);
+                    return 1;
+                }
+                if (map->map_definition.pinning == PIN_GLOBAL_NS) {
+                    char pin_path_buffer[EBPF_MAX_PIN_PATH_LENGTH];
+                    int len = snprintf(
+                        pin_path_buffer,
+                        EBPF_MAX_PIN_PATH_LENGTH,
+                        "%s/%s",
+                        pe_context->pin_root_path ? pe_context->pin_root_path : DEFAULT_PIN_ROOT_PATH,
+                        map->name);
+                    if (len < 0 || len >= EBPF_MAX_PIN_PATH_LENGTH) {
+                        clean_up_ebpf_map(map);
```

Comment:
how about go to error and do error handling there? Plus some ebpf tracing would also be helpful here.

Reply:
Done